File: tests/fixtures/go/pipeline.go
// pipeline.go — entry point composing processor and filters

package fixtures

// PipelineConfig selects the pipeline configuration
type PipelineConfig struct {
	mode string
}

// NewPipelineConfig creates a config with the given mode
func NewPipelineConfig(mode string) *PipelineConfig {
	return &PipelineConfig{mode: mode}
}

// BuildPipeline constructs the appropriate processor for the mode
func (pc *PipelineConfig) BuildPipeline() *DataProcessor {
	cfg := ProcessorConfig{RequiredFields: []string{"id", "name"}}
	proc := NewDataProcessor(cfg)
	proc.AddTransformer(NormalizeFilter{})
	proc.AddTransformer(RemoveNullFilter{})
	return proc
}

// RunPipeline processes a batch of records
func RunPipeline(inputData []map[string]interface{}) []map[string]interface{} {
	pc := NewPipelineConfig("default")
	proc := pc.BuildPipeline()
	result := make([]map[string]interface{}, len(inputData))
	for i, d := range inputData {
		result[i] = proc.Process(d)
	}
	return result
}

File: tests/fixtures/java/Pipeline.java
// Pipeline.java — entry point composing processor and filters

package fixtures;

import java.util.*;

public class Pipeline {
    static class DefaultConfig implements ProcessorConfig {
        @Override
        public List<String> getRequiredFields() {
            return Arrays.asList("id", "name");
        }

        @Override
        public boolean isStrict() {
            return false;
        }
    }

    public static DataProcessor buildPipeline(String mode) {
        ProcessorConfig config = new DefaultConfig();
        DataProcessor proc = new DataProcessor(config);
        proc.addTransformer(new NormalizeFilter());
        proc.addTransformer(new RemoveNullFilter());
        return proc;
    }

    public static List<Map<String, Object>> runPipeline(
            List<Map<String, Object>> inputData) {
        DataProcessor proc = buildPipeline("default");
        List<Map<String, Object>> result = new ArrayList<>();
        for (Map<String, Object> d : inputData) {
            result.add(proc.process(d));
        }
        return result;
    }
}
